File: ooPJAva/practice/calculator_test.py
from calculator import add, subtract, multiply, divide

def test_add_function():
    assert add(1, 2) == 3
    assert add(-1, -2) == -3

def test_subtract_function():
    assert subtract(2, 1) == 1
    assert subtract(-2, -1) == -1

def test_multiply_function():
    assert multiply(2, 3) == 6
    assert multiply(-2, -3) == 6

def test_divide_function():
    assert abs(divide(10, 3) - 3.33) < 0.01
    assert divide(-6, -3) == 2
    assert divide(1, 0) == "Error: Division by zero!"

File: ooPJAva/practice/CalculatorTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class CalculatorTest {

    @Test
    void testAddFunction() {
        assertEquals(3, Calculator.add(1, 2));
        assertEquals(-3, Calculator.add(-1, -2));
    }

    @Test
    void testSubtractFunction() {
        assertEquals(1, Calculator.subtract(2, 1));
        assertEquals(-1, Calculator.subtract(-2, -1));
    }

    @Test
    void testMultiplyFunction() {
        assertEquals(6, Calculator.multiply(2, 3));
        assertEquals(6, Calculator.multiply(-2, -3));
    }

    @Test
    void testDivideFunction() {
        assertEquals(3.33, Calculator.divide(10, 3), 0.01);
        assertEquals(2, Calculator.divide(-6, -3));
        assertEquals("Error: Division by zero!", Calculator.divide(1, 0));
    }
}
